### PR TITLE
chore: Bump Dependencies

### LINKED ISSUE #2653: bump golang.org/x/net from 0.18.0 to 0.19.0
As seen in https://github.com/kubernetes-sigs/gateway-api/pull/2638 there are some failures occurring with dependabot's attempt to update x/net. The purpose of this task is to manually get those updates done.

Similar to https://github.com/kubernetes-sigs/gateway-api/issues/2642, the reason for separating this out into its own issue rather than just doing it is that I realized this actually could probably qualify as a `good first issue`, since it mostly just requires some Golang knowledge and not deep knowledge of specific Gateway API implementations or anything like that. We never have enough `good first issues` for the amount of interest that we get in this project. So we'll hang this one out there for a minute and see if any newcomers want it first.

### PR DESCRIPTION
**Pull request by Vandit1604**
**What type of PR is this?**
chore

**What this PR does / why we need it**:
Bump golang.org/x/net from 0.18.0 to 0.19.0

**Which issue(s) this PR fixes**:

Fixes #2653 

**Does this PR introduce a user-facing change?**: NONE

Release Note:
```
Bump dependencies 

golang.org/x/net to v0.20.0
golang.org/x/net to v0.20.0
k8s.io/apiextensions-apiserver to v0.29.0
k8s.io/api to v0.29.0
k8s.io/apimachinery to v0.29.0
k8s.io/client-go to v0.29.0
k8s.io/code-generator to v0.29.0
k8s.io/utils to v0.0.0-20240102154912-e7106e64919e
sigs.k8s.io/gateway-api to v1.0.0
github.com/emicklei/go-restful/v3 to v3.11.2
github.com/go-logr/logr to v1.4.1 
github.com/go-openapi/jsonpointer to v0.20.2 
github.com/go-openapi/jsonreference to v0.20.4 
github.com/go-openapi/swag to v0.22.7 
github.com/google/uuid to v1.5.0 
github.com/gorilla/websocket to v1.5.1 
github.com/mxk/go-flowrate to v0.0.0-20140419014527-cca7078d478f 
golang.org/x/mod to v0.14.0 
golang.org/x/oauth2  to v0.16.0 
golang.org/x/sys to v0.16.0 
golang.org/x/term to v0.16.0 
golang.org/x/time to v0.5.0 
golang.org/x/tools to v0.16.1 
google.golang.org/protobuf  to v1.32.0 
k8s.io/klog/v2 to v2.120.0 
k8s.io/kube-openapi to v0.0.0-20240105020646-a37d4de58910 
sigs.k8s.io/structured-merge-diff/v4 to v4.4.1 
golang.org/x/net to v0.20.0 
golang.org/x/sys to v0.16.0 
golang.org/x/term to v0.16.0
github.com/spf13/cobra to v1.8.0
```

### CODE REVIEW

**Comment by Vandit1604** [comment]
/assign robscott
Assigning according to the CI bot

**Review by hzxuzhonghu — COMMENTED** [review]
/lgtm

**Comment by Vandit1604** [comment]
Hey, Is there anything else that needs to be done for this PR so we can merge it?

File: conformance/echo-basic/go.mod [hunk 1]
```diff
@@ -2,6 +2,6 @@ module sigs.k8s.io/gateway-api/conformance/echo-basic
 
 go 1.21
 
-require golang.org/x/net v0.17.0
+require golang.org/x/net v0.19.0
```
Comment (robscott) [inline conformance/echo-basic/go.mod:5]: Why is this getting bumped to v0.19.0 while the root go.mod is getting bumped to v0.20.0?

**Review by robscott — COMMENTED** [review]
Thanks @Vandit1604!

File: conformance/echo-basic/go.mod (hunk 1 above)
Comment (Vandit1604) [reply conformance/echo-basic/go.mod]: I have addressed the inconsistencies in the latest commit.

Comment (robscott) [reply conformance/echo-basic/go.mod]: Thanks @Vandit1604! Do you mind updating the PR title and release note to include the version the PR is bumping everything to now?

Comment (Vandit1604) [reply conformance/echo-basic/go.mod]: Done.

**Comment by robscott** [comment]
Thanks @Vandit1604! LGTM once you can get tests passing (should just need a `make generate` run).

/approve

**Comment by Vandit1604** [comment]
Hey @robscott Idk why but `make generate` changes this file 

```
root@vandit-box:/home/vandit/go/src/sigs.k8s.io/gateway-api# git status
On branch chore/bump-dependencies
Your branch is up to date with 'origin/chore/bump-dependencies'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   pkg/client/informers/externalversions/factory.go
```

These are the changes 

```
root@vandit-box:/home/vandit/go/src/sigs.k8s.io/gateway-api# git diff
diff --git a/pkg/client/informers/externalversions/factory.go b/pkg/client/informers/externalversions/factory.go
index f4479aa1..e7f08e89 100644
--- a/pkg/client/informers/externalversions/factory.go
+++ b/pkg/client/informers/externalversions/factory.go
@@ -42,6 +42,7 @@ type sharedInformerFactory struct {
        lock             sync.Mutex
        defaultResync    time.Duration
        customResync     map[reflect.Type]time.Duration
+       transform        cache.TransformFunc
 
        informers map[reflect.Type]cache.SharedIndexInformer
        // startedInformers is used for tracking which informers have been started.
@@ -80,6 +81,14 @@ func WithNamespace(namespace string) SharedInformerOption {
        }
 }
 
+// WithTransform sets a transform on all informers.
+func WithTransform(transform cache.TransformFunc) SharedInformerOption {
+       return func(factory *sharedInformerFactory) *sharedInformerFactory {
+               factory.transform = transform
+               return factory
+       }
+}
+
 // NewSharedInformerFactory constructs a new instance of sharedInformerFactory for all namespaces.
 func NewSharedInformerFactory(client versioned.Interface, defaultResync time.Duration) SharedInformerFactory {
        return NewSharedInformerFactoryWithOptions(client, defaultResync)
@@ -184,6 +193,7 @@ func (f *sharedInformerFactory) InformerFor(obj runtime.Object, newFunc internal
        }
 
        informer = newFunc(f.client, resyncPeriod)
+       informer.SetTransform(f.transform)
        f.informers[informerType] = informer
 
        return informer
```

Due to which the test is failing while running `make` passes every test.

**Comment by robscott** [comment]
Hey @Vandit1604, have you tried committing those changes to get CI to pass? It is not uncommon for generated code to change when we upgrade k8s dependencies.

**Comment by Vandit1604** [comment]
> Hey @Vandit1604, have you tried committing those changes to get CI to pass? It is not uncommon for generated code to change when we upgrade k8s dependencies.

I have committed the changes but I have a question if you may, 
What's the auto-generated code by the CI and How does it generated. If you could just point me to the code by which it happen. I would like to take a look.

**Comment by Vandit1604** [comment]
> Hey @Vandit1604, have you tried committing those changes to get CI to pass? It is not uncommon for generated code to change when we upgrade k8s dependencies.

I've committed the changes, but if I may ask, what is the CI's auto-generated code and how is it generated? If you could just point me to the code that generates it, that would be great. I'd like to have a look.

**Comment by robscott** [comment]
> I've committed the changes, but if I may ask, what is the CI's auto-generated code and how is it generated? If you could just point me to the code that generates it, that would be great. I'd like to have a look.

Hey @Vandit1604, great question! That code generation happens here:

https://github.com/kubernetes-sigs/gateway-api/blob/35fe25d1384a41c9b89dd5af7ae3214c431f008c/hack/update-codegen.sh#L64-L83

**Comment by robscott** [comment]
Thanks @Vandit1604!

/lgtm
/release-note-none

**Pull request merged**